portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,23 +51,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<Post> posts = entityToList(results, request);
     
     response.setContentType("application/json;");
+    response.setCharacterEncoding("UTF-8");
     response.getWriter().println(convertToJson(posts));
   }
   
   // Posts information to the server to save as inputs from the data form
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Entity commentEntity = createEntity(request);
-
+  
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect("/contact.html");
   }
+  
+  private String getLanguage(HttpServletRequest request) {
+    String languageChoice;
+    try {
+      languageChoice = request.getParameter("language");
+    } catch (NullPointerException e) {
+      return DEFAULT_LANGUAGE;
+    }
+    return languageChoice;
+  }
 
   private int getCommentNum(HttpServletRequest request) {
-    String numChoiceString = request.getParameter("quantity");
-   
+    String numChoiceString;
+    try {
+      numChoiceString = request.getParameter("quantity");
+    } catch (NullPointerException e) {
```

Comment:
Same comment as above. Also, I've noticed this logic is repeated a few times. Perhaps refactor it into a method and call that?

Reply:
I tried to refactor these, but I was having some trouble. Since the parameters return different types, I was getting a lot of different type mismatch errors and ultimately ended up adding a lot of extra code that seemed even more redundant to me. Even through the structure is similar, the function calls and caught errors are different enough that I could not think of an efficient refactoring. Do you have any thoughts?

Reply:
Can you pass the exception as a parameter? You may need to opt for the base exception class here. To handle the varying method, there is a way to pass functions as parameters in Java.

Reply:
It seems like the only way to pass a function as a parameter would be to create a separate interface and call the function through that. I feel like there should be a much simpler way of doing this since its only 5 lines of code -- I'm going to think through this a little bit more throughout the day, but I don't have a great solution right now. In terms of passing exceptions as a parameter, I don't think that's possible, but that also isn't a big issue because all of the exceptions that I'm using are unchecked.

Reply:
Resolved offline. This comment is not a blocker to this PR. Glad to help you see others avenues for this.